### PR TITLE
Print results in tabulated format

### PR DESCRIPTION
**Pull request by jianli**

### CODE REVIEW

File: cmd/dosa/util.go [hunk 1]
```diff
@@ -114,16 +117,35 @@ func strToFieldValue(t dosa.Type, s string) (dosa.FieldValue, error) {
 	}
 }
 
-func printResult(res []map[string]dosa.FieldValue) {
-	if len(res) == 0 {
-		fmt.Println("Error: not found")
-		return
+func printResults(results []map[string]dosa.FieldValue) error {
+	const padding = 3
```
Comment (MaxJTRice) [inline cmd/dosa/util.go]: Should this be a constant? If so maybe it's better to define outside this function (under the imports)?

File: cmd/dosa/util.go [hunk 2]
```diff
@@ -114,16 +117,35 @@ func strToFieldValue(t dosa.Type, s string) (dosa.FieldValue, error) {
...
+	w := tabwriter.NewWriter(os.Stdout, 0, 0, padding, ' ', tabwriter.Debug)
+	if len(results) == 0 {
+		return errors.New("Empty results")
```
Comment (MaxJTRice) [inline cmd/dosa/util.go]: You may need to fix a test for this error message :)

File: cmd/dosa/util.go [hunk 3]
```diff
@@ -114,16 +117,35 @@ func strToFieldValue(t dosa.Type, s string) (dosa.FieldValue, error) {
...
-			fmt.Printf("%s: %v\n", field, reflect.Indirect(reflect.ValueOf(value)))
+	i := 0
+	width := len(results[0])
+	keys := make([]string, width)
+	for k := range results[0] {
+		keys[i] = k
+		i++
+	}
+	sort.Strings(keys)
+	_, err := fmt.Fprintln(w, strings.Join(keys, "\t"))
```
Comment (MaxJTRice) [inline cmd/dosa/util.go]: nit: you can have the same logic like this:

if _, err := fmt.Fprintln(w, strings.Join(keys, "\t")); err != nil {
    return errors.WithStack(err)
}

File: cmd/dosa/util.go [hunk 4]
```diff
@@ -114,16 +117,35 @@ func strToFieldValue(t dosa.Type, s string) (dosa.FieldValue, error) {
...
+		return errors.WithStack(err)
+	}
+	values := make([]string, width)
+	for _, result := range results {
+		i := 0
+		for _, key := range keys {
+			values[i] = fmt.Sprintf("%v", reflect.Indirect(reflect.ValueOf(result[key])))
+			i++
+		}
+		_, err := fmt.Fprintln(w, strings.Join(values, "\t"))
```
Comment (MaxJTRice) [inline cmd/dosa/util.go]: nit: same as ^^

File: cmd/dosa/util.go [hunk 5]
```diff
@@ -114,16 +117,35 @@ func strToFieldValue(t dosa.Type, s string) (dosa.FieldValue, error) {
...
+	}
+	sort.Strings(keys)
+	_, err := fmt.Fprintln(w, strings.Join(keys, "\t"))
+	if err != nil {
+		return errors.WithStack(err)
+	}
+	values := make([]string, width)
+	for _, result := range results {
+		i := 0
+		for _, key := range keys {
```
Comment (MaxJTRice) [inline cmd/dosa/util.go]: nit: here maybe we can write like
'for idx, result := range results'
and get rid of i := 0 and i++?

File: cmd/dosa/util.go [hunk 6]
```diff
@@ -114,16 +117,35 @@ func strToFieldValue(t dosa.Type, s string) (dosa.FieldValue, error) {
...
+	const padding = 3
+	w := tabwriter.NewWriter(os.Stdout, 0, 0, padding, ' ', tabwriter.Debug)
+	if len(results) == 0 {
+		return errors.New("Empty results")
 	}
-	fmt.Println("Results:")
-	for _, fieldValues := range res {
-		for field, value := range fieldValues {
-			fmt.Printf("%s: %v\n", field, reflect.Indirect(reflect.ValueOf(value)))
+	i := 0
```
Comment (MaxJTRice) [inline cmd/dosa/util.go]: nit: maybe it's better to have a meaningful parameter name rather than 'i'? :)

File: cmd/dosa/query.go [hunk 7]
```diff
@@ -99,7 +99,10 @@ func (c *QueryCmd) doQueryOp(f func(ShellQueryClient, context.Context, []*queryO
 		return err
 	}
 
-	printResult(results)
+	err = printResults(results)
```
Comment (junchaowu) [inline cmd/dosa/query.go]: put it in one line

**Comment by jianli** [comment]
@rkuris I've added unit tests as well as escaping for whitespace characters.

Our use case is quickly getting DOSA rows for visual inspection (in the same way that you would query any standard SQL interface) so we don't need HTML or JSON output.

File: cmd/dosa/util.go [hunk 8]
```diff
@@ -114,16 +117,35 @@ func strToFieldValue(t dosa.Type, s string) (dosa.FieldValue, error) {
...
+		return errors.New("Empty results")
 	}
-	fmt.Println("Results:")
-	for _, fieldValues := range res {
-		for field, value := range fieldValues {
-			fmt.Printf("%s: %v\n", field, reflect.Indirect(reflect.ValueOf(value)))
+	idx := 0
+	width := len(results[0])
+	keys := make([]string, width)
+	for k := range results[0] {
```
Comment (MaxJTRice) [inline cmd/dosa/util.go]: Found 1 issue from my tests here when there's empty (null) value for the results:
e.g. the table has three columns (c1, c2, c3), we insert two rows: 

row1: (r1c1, nil, r1c3)
row2: (r2c1, r2c2, r2c3)

Two possibilities:
1. If row1 is results[0], then the CLI will only show columns c1 and c3
2. If row2 is results[0], then the CLI will show columns c1, c2 and c3, but row1's c2 value will be: <invalid reflect.Value>

Suggestions:
1. For printResults function, you can pass the list of fields to show (add fields to input signature). You can get the fields from: https://github.com/uber-go/dosa/blob/master/cmd/dosa/query.go#L92
2. We check if result[key] is valid. If not, we print 'nil' for it like cqlsh does.

**Pull request merged**